### ISSUE TITLE
linter: Printed Default Config and JSON/Markdown Rules Lists Disagree on `eslint/no-implied-eval` Enabled/Default Setting

### ISSUE THREAD

**Issue by bachmacintosh** [oxc-project/oxc#26209]
### What version of Oxlint are you using?

1.80.0

### What command did you run?

`oxlint --rules`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
// Default config on oxlint 1.80.0 when no config file is discovered or specified per `oxlint --print-config`
{
  "plugins": [
    "unicorn",
    "typescript",
    "oxc"
  ],
  "categories": {},
  "rules": {
    "constructor-super": "warn",
    "for-direction": "warn",
    "getter-return": "warn",
    "no-async-promise-executor": "warn",
    "no-caller": "warn",
    "no-class-assign": "warn",
    "no-compare-neg-zero": "warn",
    "no-cond-assign": "warn",
    "no-const-assign": "warn",
    "no-constant-binary-expression": "warn",
    "no-constant-condition": "warn",
    "no-control-regex": "warn",
    "no-debugger": "warn",
    "no-delete-var": "warn",
    "no-dupe-class-members": "warn",
    "no-dupe-else-if": "warn",
    "no-dupe-keys": "warn",
    "no-duplicate-case": "warn",
    "no-empty-character-class": "warn",
    "no-empty-pattern": "warn",
    "no-empty-static-block": "warn",
    "no-eval": "warn",
    "no-ex-assign": "warn",
    "no-extra-boolean-cast": "warn",
    "no-func-assign": "warn",
    "no-global-assign": "warn",
    "no-import-assign": "warn",
    "no-invalid-regexp": "warn",
    "no-irregular-whitespace": "warn",
    "no-iterator": "warn",
    "no-loss-of-precision": "warn",
    "no-misleading-character-class": "warn",
    "no-new-native-nonconstructor": "warn",
    "no-nonoctal-decimal-escape": "warn",
    "no-obj-calls": "warn",
    "no-self-assign": "warn",
    "no-setter-return": "warn",
    "no-shadow-restricted-names": "warn",
    "no-sparse-arrays": "warn",
    "no-this-before-super": "warn",
    "no-unassigned-vars": "warn",
    "no-unreachable": "warn",
    "no-unsafe-finally": "warn",
    "no-unsafe-negation": "warn",
    "no-unsafe-optional-chaining": "warn",
    "no-unused-expressions": "warn",
    "no-unused-labels": "warn",
    "no-unused-private-class-members": "warn",
    "no-unused-vars": "warn",
    "no-useless-backreference": "warn",
    "no-useless-catch": "warn",
    "no-useless-escape": "warn",
    "no-useless-rename": "warn",
    "no-with": "warn",
    "require-yield": "warn",
    "use-isnan": "warn",
    "valid-typeof": "warn",
    "oxc/bad-array-method-on-arguments": "warn",
    "oxc/bad-char-at-comparison": "warn",
    "oxc/bad-comparison-sequence": "warn",
    "oxc/bad-match-all-arg": "warn",
    "oxc/bad-min-max-func": "warn",
    "oxc/bad-object-literal-comparison": "warn",
    "oxc/bad-replace-all-arg": "warn",
    "oxc/const-comparisons": "warn",
    "oxc/double-comparisons": "warn",
    "oxc/erasing-op": "warn",
    "oxc/missing-throw": "warn",
    "oxc/number-arg-out-of-range": "warn",
    "oxc/only-used-in-recursion": "warn",
    "oxc/uninvoked-array-callback": "warn",
    "typescript/await-thenable": "warn",
    "typescript/no-array-delete": "warn",
    "typescript/no-base-to-string": "warn",
    "typescript/no-duplicate-enum-values": "warn",
    "typescript/no-duplicate-type-constituents": "warn",
    "typescript/no-extra-non-null-assertion": "warn",
    "typescript/no-floating-promises": "warn",
    "typescript/no-for-in-array": "warn",
    "typescript/no-implied-eval": "warn",
    "typescript/no-meaningless-void-operator": "warn",
    "typescript/no-misused-new": "warn",
    "typescript/no-misused-spread": "warn",
    "typescript/no-non-null-asserted-optional-chain": "warn",
    "typescript/no-redundant-type-constituents": "warn",
    "typescript/no-this-alias": "warn",
    "typescript/no-unnecessary-parameter-property-assignment": "warn",
    "typescript/no-unsafe-declaration-merging": "warn",
    "typescript/no-unsafe-unary-minus": "warn",
    "typescript/no-useless-default-assignment": "warn",
    "typescript/no-useless-empty-export": "warn",
    "typescript/no-wrapper-object-types": "warn",
    "typescript/prefer-as-const": "warn",
    "typescript/prefer-namespace-keyword": "warn",
    "typescript/require-array-sort-compare": "warn",
    "typescript/restrict-template-expressions": "warn",
    "typescript/triple-slash-reference": "warn",
    "typescript/unbound-method": "warn",
    "unicorn/no-await-in-promise-methods": "warn",
    "unicorn/no-empty-file": "warn",
    "unicorn/no-invalid-fetch-options": "warn",
    "unicorn/no-invalid-remove-event-listener": "warn",
    "unicorn/no-new-array": "warn",
    "unicorn/no-single-promise-in-promise-methods": "warn",
    "unicorn/no-thenable": "warn",
    "unicorn/no-unnecessary-await": "warn",
    "unicorn/no-useless-fallback-in-spread": "warn",
    "unicorn/no-useless-length-check": "warn",
    "unicorn/no-useless-spread": "warn",
    "unicorn/prefer-set-size": "warn",
    "unicorn/prefer-string-starts-ends-with": "warn"
  },
  "settings": {
    "jsx-a11y": {
      "polymorphicPropName": null,
      "components": {},
      "attributes": {}
    },
    "next": {
      "rootDir": []
    },
    "react": {
      "formComponents": [],
      "linkComponents": [],
      "version": null,
      "componentWrapperFunctions": []
    },
    "jsdoc": {
      "ignorePrivate": false,
      "ignoreInternal": false,
      "ignoreReplacesDocs": true,
      "overrideReplacesDocs": true,
      "augmentsExtendsReplacesDocs": false,
      "implementsReplacesDocs": false,
      "exemptDestructuredRootsFromChecks": false,
      "tagNamePreference": {}
    },
    "vitest": {
      "typecheck": false
    },
    "jest": {
      "version": null
    }
  },
  "env": {
    "builtin": true
  },
  "globals": {},
  "ignorePatterns": []
}
```

### What happened?

I'm in the process of building a shared config for oxlint, and while going through the rules reference in the docs, I noticed that [eslint/no-implied-eval](https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-implied-eval) was shown as enabled by default in the docs, including on [the suspicious eslint rules page](https://oxc.rs/docs/guide/usage/linter/rules.html?sort=name&dir=asc&scope=eslint&category=suspicious). This felt contrary to other language in the docs that implied that only `correctness` rules are enabled by default.

I have oxlint installed in the shared config project that I'm working on alongside TypeScript 7 and tsgolint as other devDependencies, and renamed the project's config so that oxlint wouldn't discover it. I used `oxlint --print-config` to get the default config when no config file is discovered or specified, and noticed `no-implied-eval` wasn't set in the outputted config while `typescript/no-implied-eval` was set to `warn`. I then ran `oxlint --rules`, and the markdown output came back with the following (which has been truncated for easier reading):

```markdown
## Correctness (272)
Code that is outright wrong or useless.
| Rule name                                           | Source     | Default | Enabled? | Fixable? |
| --------------------------------------------------- | ---------- | ------- | -------- | -------- |
| no-implied-eval                                     | typescript | ✅      | ✅       |          |

## Suspicious (63)
code that is most likely wrong or useless.
| Rule name                              | Source     | Default | Enabled? | Fixable? |
| -------------------------------------- | ---------- | ------- | -------- | -------- |
| no-implied-eval                        | eslint     | ✅      | ✅       |          |
```

When I ran `oxlint --rules --format=json` to see if this was isolated to a specific output format, I received similar output (again truncated for easier reading):

```jsonc
[
  {
    "scope": "eslint",
    "value": "no-implied-eval",
    "category": "suspicious",
    "type_aware": false,
    "fix": "none",
    "default": true,
    "docs_url": "https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-implied-eval.html"
  },
  {
    "scope": "typescript",
    "value": "no-implied-eval",
    "category": "correctness",
    "type_aware": true,
    "fix": "none",
    "default": true,
    "docs_url": "https://oxc.rs/docs/guide/usage/linter/rules/typescript/no-implied-eval.html"
  }
]
```

There is likely a rule name collision occurring where the `default_rules` set for the Markdown and JSON formatting is getting only the rule name (eg `no-implied-eval` for both the `eslint` and `typescript` versions), and when constructing a `default` or `turned_on_by_default` value for the output, it only has the name to go by, not the exact plugin.

https://github.com/oxc-project/oxc/blob/2436b4407c03939083219d99100ffd1e7f9442c7/crates/oxc_linter/src/table.rs#L49-L77

https://github.com/oxc-project/oxc/blob/2436b4407c03939083219d99100ffd1e7f9442c7/apps/oxlint/src/output_formatter/json.rs#L38-L65

One approach might be to include both the plugin name and rule name when constructing the value on those sets, but I don't have enough familiarity with the project to say that's the best course of action relevant to anything downstream of this formatting logic (besides docs generation).

I'm not entirely certain as to why `eslint/no-implied-eval` is showing as Enabled in the outputted Markdown, and wouldn't want to speculate a deduction like can be seen with the "default" logic.

I'm happy to provide any additional needed info. Thanks for looking into it!

## AI Use Disclosure

I used Claude Opus 5 to help locate the two cited files within the codebase that pertained to outputting rules lists when running `oxlint --rules`. Everything else -- setting up/running oxlint in its default state within a project, analyzing/deducing the possible issue within the rule table/output formatting logic, and writing the issue itself -- was done by myself.
